spring-cloud-dataflow-docs/src/main/asciidoc/architecture.adoc

```diff
@@ -0,0 +1,223 @@
+[[architecture]]
+= Architecture
+
+[[arch-intro]]
+== Introduction
+
+Spring Cloud Data Flow simplifies the development and deployment of applications focused on data processing use-cases.  The major concepts of the architecture are Applications, the Data Flow Server, and the target runtime.
+
+ifdef::omit-tasks-docs[]
+Applications are Long lived Stream applications where an unbounded amount of data is consumed or produced via messaging middleware.
+endif::omit-tasks-docs[]
+ifndef::omit-tasks-docs[]
+Applications come in two flavors
+
+* Long lived Stream applications where an unbounded amount of data is consumed or produced via messaging middleware.
+* Short lived Task applications that process a finite set of data and then terminate.
+endif::omit-tasks-docs[]
+
+Depending on the runtime, applications can be packaged in two ways
+
+* Maven
```

Comment:
we should instead say it is Spring boot uber-jar in the bullet list and then mention in the text that it can be hosted in a maven repository or file,http. - or any spring resource implementation